tests/sanity/extra/action-group.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+# Copyright (c) 2024, Felix Fontein <felix@fontein.de>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+"""Make sure all modules that should show up in the action group."""
+
+from __future__ import annotations
+
+import os
+import re
+import yaml
+
+
+ACTION_GROUPS = {
+    # The format is as follows:
+    # * 'pattern': a regular expression matching all module names potentially belonging to the action group;
+    # * 'exclusions': a list of modules that are not part of the action group; all other modules matching 'pattern' must be part of it;
+    # * 'doc_fragment': the docs fragment that documents membership of the action group.
+    'consul': {
+        'pattern': re.compile('^consul_.*$'),
+        'exclusions': [
+            'consul_acl_bootstrap',
+            'consul_kv',
+        ],
+        'doc_fragment': 'community.general.consul.actiongroup_consul',
+    },
+    'proxmox': {
+        'pattern': re.compile('^proxmox(|_.*)$'),
```

Comment:
I'm OK with using `(_.*)?` instead of `(|_.*)`, but I don't think `(_\w+)?` is a good idea. That for example prevents `proxmox_module_with_more_underscores`. The pattern should simply match everything with prefix `proxmox_`, and the legacy name `proxmox` as well.

Reply:
```suggestion
        'pattern': re.compile('^proxmox(_.*)?$'),
```

Reply:
Actually, ... first, I beg to differ: `\w` is equivalent to `[A-Za-z0-9_]` so it already has the `_` in it. Which means that I should have simplified it further:
```suggestion
        'pattern': re.compile('^proxmox\w*?$'),
```
That's all we need. ;-)

Reply:
Ah, I didn't remember `\w` also includes `_`. I wouldn't call an underscore a word character, which is what `\w` stands for, but that ship sailed a *long* time ago...